src/rgw/rgw_rest_iam_policy.cc

```diff
@@ -0,0 +1,384 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab ft=cpp
+
+#include "rgw_rest_iam_policy.h"
+#include "rgw_rest_iam.h"
+#include <regex>
+#include <string>
+#include "rgw_process_env.h"
+#include <errno.h>
+
+
+#define dout_subsys ceph_subsys_rgw
+
+int RGWRestPolicy::verify_permission(optional_yield y)
+{
+  if (verify_user_permission(this, s, resource, action)) {
+    return 0;
+  }
+  return -EACCES;
+}
+
+void RGWRestPolicy::send_response()
+{
+  if (op_ret) {
+    set_req_state_err(s, op_ret);
+  }
+  dump_errno(s);
+  end_header(s, this);
+}
+
+static int parse_tags(const DoutPrefixProvider* dpp,
+                      const std::map<std::string, std::string>& params,
+                      std::multimap<std::string, std::string>& tags,
+                      std::string& message)
+{
+  std::map<int, std::string> key_map;
+  std::map<int, std::string> val_map;
+
+  const std::regex pattern_key("Tags\\.member\\.([0-9]+)\\.Key");
+  const std::regex pattern_value("Tags\\.member\\.([0-9]+)\\.Value");
+
+  for (const auto& param : params) {
+    std::smatch match;
+
+    if (std::regex_match(param.first, match, pattern_key)) {
+      int index = std::stoi(match[1].str());
+      key_map[index] = param.second;
+    }
+
+    if (std::regex_match(param.first, match, pattern_value)) {
+      int index = std::stoi(match[1].str());
```

Comment:
[std::stoi()](https://en.cppreference.com/w/cpp/string/basic_string/stol) can throw exceptions that would need to be caught. consider using `ceph::parse<int>()` from `common/strtol.h` instead - it's based on c++17's [std::from_chars()](https://en.cppreference.com/w/cpp/utility/from_chars) which doesn't throw